app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  belongs_to :users, class_name: 'User', foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'post_id'
+  has_many :comments, foreign_key: 'post_id'
+
+  after_create :update_posts_counter
+
+  def update_posts_counter
+    author.increment!(:posts_count)
+  end
```

Comment:
- [ ] Hi, I noticed that you named the variable `user` and used it as `author` in the Post model. This is not a good practice because it does not follow the Rails convention of naming associations. It also makes the code harder to read and understand, because user and author are not the same thing. A post belongs to an author, not a user. A user can be an author of many posts, but a post can only have one author.
I suggest that you rename the variable user to author and use it consistently throughout the model. You can also use the shorthand syntax for specifying the class name and foreign key of the association. For example, instead of writing this:

```rb
class Post < ApplicationRecord
  belongs_to :users, class_name: 'User', foreign_key: 'author_id'
  # ...
end
```

You can write this:

```rb
class Post < ApplicationRecord
  belongs_to :author, class_name: 'User'
  # ...
end
```